test/integration/httproute_test.go

```diff
@@ -596,13 +597,22 @@ func TestHTTPRouteFilterHosts(t *testing.T) {
 	require.False(t, testGetByHost(t, "another.specific.io"))
 
 	t.Logf("update hostnames in httproute to wildcard")
-	httpRoute, err = gatewayClient.GatewayV1beta1().HTTPRoutes(ns.Name).Get(ctx, httpRoute.Name, metav1.GetOptions{})
-	require.NoError(t, err)
-	httpRoute.Spec.Hostnames = []gatewayv1beta1.Hostname{
-		gatewayv1beta1.Hostname("*.specific.io"),
-	}
-	_, err = gatewayClient.GatewayV1beta1().HTTPRoutes(ns.Name).Update(ctx, httpRoute, metav1.UpdateOptions{})
-	require.NoError(t, err)
+	require.Eventually(t, func() bool {
+		httpRoute, err = hClient.Get(ctx, httpRoute.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("failed getting the HTTPRoute %s: %v", httpRoute.Name, err)
+			return false
+		}
+		httpRoute.Spec.Hostnames = []gatewayv1beta1.Hostname{
+			gatewayv1beta1.Hostname("*.specific.io"),
+		}
+		httpRoute, err = hClient.Update(ctx, httpRoute, metav1.UpdateOptions{})
```

Comment:
Seems that repeated query still fails https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3914699729/jobs/6692057063. Will `Patch` solve the problem?

Reply:
🤔 😠 

The Update didn't fail though because there are no logs in the linked CI output and the failure is from the `.Evantually()` block below. So it was applied ( and `.Update()` contrary to `.Patch()` updates the whole object, it does not patch it).

There has to be some other problem that this approach as it stands doesn't fix 🤔